### PR TITLE
Check whether a number is a Perfect Number or not in C++

### PR DESCRIPTION
**Pull request by Aayushi-Mittal**
…or not

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

A program to check whether a number is perfect number or not in C++.
Perfect number is a positive integer which is equal to the sum of its proper positive divisors.
For example: 6 is the first perfect number
Proper divisors of 6 are 1, 2, 3. Sum of its proper divisors = 1 + 2 + 3 = 6. Hence 6 is a perfect number.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4853

### CODE REVIEW

**Comment by Aayushi-Mittal** [comment]
@HarshCasper @siddharth25pandey  Please review my PR.

File: C-Plus-Plus/math/Perfect_Number.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    return 0;
+}
+
+/*
+Time Complexity: O(n)
+Space Complexity: O(1)
+
+sample 1:
+Input: Enter a number to check whether it is a perfect number or not: 6
+Output: 6 is a perfect number.
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/Perfect_Number.cpp:42]: Add a new line after output

File: C-Plus-Plus/math/Perfect_Number.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
+Perfect number is a positive integer which is equal to the sum of its proper positive divisors.
+For example: 6 is the first perfect number
+Proper divisors of 6 are 1, 2, 3
+Sum of its proper divisors = 1 + 2 + 3 = 6.
+Hence 6 is a perfect number.
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/Perfect_Number.cpp:10]: Add a new line above int main

File: C-Plus-Plus/math/Perfect_Number.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        if (n % i == 0)
+            sum += i;
+    }
+    if (sum == n)
+        cout << sum << " is a perfect number." << endl;
+    else
+        cout << n << " is not a perfect number." << endl;
+    return 0;
+}
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/Perfect_Number.cpp]: Shift the logic to another function and use main for driver code

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Aayushi-Mittal** [comment]
@ankitaggarwal23  Made the required changes.

File: C-Plus-Plus/math/Perfect_Number.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+For example: 6 is the first perfect number
+Proper divisors of 6 are 1, 2, 3
+Sum of its proper divisors = 1 + 2 + 3 = 6.
+Hence 6 is a perfect number.
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+
+int sumOfDivisors(int n)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/Perfect_Number.cpp]: Rename it to perfectnumber

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Aayushi-Mittal** [comment]
@ankitaggarwal23 did the changes. Please review.

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by Aayushi-Mittal** [comment]
Thank you so much!

**Pull request merged**